documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -45,6 +45,7 @@ in the `junit-jupiter-api` module.
 | `@ExtendWith`            | Used to <<extensions-registration-declarative,register extensions declaratively>>. Such annotations are _inherited_.
 | `@RegisterExtension`     | Used to <<extensions-registration-programmatic,register extensions programmatically>> via fields. Such fields are _inherited_ unless they are _shadowed_.
 | `@TempDir`               | Used to supply a <<writing-tests-built-in-extensions-TempDirectory,temporary directory>> via field injection or parameter injection in a lifecycle method or test method; located in the `org.junit.jupiter.api.io` package.
+| `@AutoClose`             | Denotes that the annotated field represents a resource that should be automatically closed after test execution.
```

Comment:
Suggest (if I've understood correctly):
```suggestion
| `@AutoClose`             | Denotes that the annotated field represents a resource that will be automatically closed after each test execution.
```

Reply:
Not necessarily. It depends on the scope of the field and the test instance lifecycle.

A closeable instance field will be closed after each test execution when the test instance lifecycle is "per test method".

But for test instance lifecycle "per test class", the closeable field will only be closed when the test instance is destroyed.

Reply:
Though we can improve the wording to hint at "scope" rules and ensure the scope rules are clearly covered in the Javadoc.

Reply:
This has been addressed in my [branch](https://github.com/junit-team/junit5/compare/main...sbrannen:junit5:issues/3592-AutoClose).